### PR TITLE
Creation of a service for request dialogs

### PR DESCRIPTION
**Pull request by ruanres**
**Feature/Bug description:** I created this service to be responsible for providing the appropriate dialog, depending on its type, to notifications and other controllers that need to deal with such dialogs.

**Solution:** ...

**TODO/FIXME:** n/a

### CODE REVIEW

File: frontend/requests/request_processing.html [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <md-dialog flex-gt-sm="{{requestCtrl.getSizeGtSmDialog()}}" flex-sm="80" flex-xs="100">
-  <div ng-if="!requestCtrl.warnPaternityExistence">
     <md-dialog-content ng-if="!requestCtrl.isRejecting" class="md-dialog-content" layout-gt-sm="column" layout-padding>
       <h4 style="margin: 0;">Confirmar Vínculo</h4>
       <p style="margin: 0;">
-        {{requestCtrl.isRequestUser() ? "Um usuário da plataforma" : "Uma instituição"}} solicitou vincular-se a uma das instituições que você administra. Clique em confirmar para aceitar ou rejeitar para recusar vinculo.
+        {{requestCtrl.isRequestUser() ? "Um usuário da plataforma" : "Uma instituição"}} solicitou vincular-se a uma das instituições que você administra.
```
Comment (mayzabeel) [inline frontend/requests/request_processing.html]: change name of html to be more specific to user request to be member

File: frontend/invites/inviteInstHierarchieController.js [hunk 2]
```diff
@@ -355,11 +355,11 @@
...
 
         inviteInstHierCtrl.analyseRequest = function analyseRequest(event, request) {
-            RequestInvitationService
-                .analyseReqDialog(event, inviteInstHierCtrl.institution, request)
+            RequestDialogService
+                .showHierarchyDialog(request, event)
                 .then(function accepted() {
                     addInstitutionToHierarchy(request);
-                });
+                }, function rejected(response) {});
```
Comment (raonismaneoto) [inline frontend/invites/inviteInstHierarchieController.js]: this rejected function is useless, isn't it?

File: frontend/requests/requestDialogService.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        service.showRequestDialog = function showRequestDialog(notification, event, dialogProperties) {
+            getRequest(notification.entity.key, notification.entity_type).then(
+                function success(data) {
+                    var request = new Invite(data);
+                    selectDialogToShow(request, event, dialogProperties);
+                }, function error(response) {
+                    MessageService.showToast(response.data.msg);
+                }
+            );
+        }
```
Comment (raonismaneoto) [inline frontend/requests/requestDialogService.js]: insert ;

File: frontend/requests/requestDialogService.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            
+            if(isRequestResolved) {
+                service.showResolvedReqDialog(event);
+                return;
+            }
+
+            switch(request.type_of_invite) {
+                case "REQUEST_INSTITUTION_PARENT":
+                case "REQUEST_INSTITUTION_CHILDREN":
+                    service.showHierarchyDialog(request, event)
```
Comment (raonismaneoto) [inline frontend/requests/requestDialogService.js]: insert ;

File: frontend/requests/requestDialogService.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+            $mdDialog.show({
+                templateUrl: "app/requests/resolved_request_dialog.html",
+                controller: ResolvedRequesCtrl,
+                controllerAs: 'ctrl',
+                parent: angular.element(document.body),
+                targetEvent: event,
+                clickOutsideToClose:true
+            });
+        }
```
Comment (raonismaneoto) [inline frontend/requests/requestDialogService.js]: ;

File: frontend/requests/requestDialogService.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                controllerAs: dialogProperties.controllerAs,
+                templateUrl: dialogProperties.templateUrl,
+                parent: angular.element(document.body),
+                targetEvent: event,
+                clickOutsideToClose:true,
+                locals: dialogProperties.locals,
+                openFrom: '#fab-new-post',
+                closeTo: angular.element(document.querySelector('#fab-new-post'))
+            });
+        }
```
Comment (raonismaneoto) [inline frontend/requests/requestDialogService.js]: ;

File: frontend/test/specs/requestProcessingControllerSpec.js [hunk 7]
```diff
@@ -293,83 +293,4 @@
             expect(requestCtrl.warnPaternityExistence).toBeFalsy();
```
Comment (raonismaneoto) [inline frontend/test/specs/requestProcessingControllerSpec.js]: remove it and every test that checks warnPaternityExistence once you've removed this flag...

**Pull request merged**